config/config_test.go

```diff
@@ -23,33 +23,14 @@ func TestEnvVarOverrides(t *testing.T) {
 	yamlString, _ := Marshal(conf)
 	conf, _ = Unmarshal(yamlString)
 
-	// we don't have the env vars set yet - so nothing should be overridden from the original yaml
+	// we don't have the files yet - so nothing should be overridden from the original yaml
 	assert.Equal(t, conf.ExternalServices.Grafana.Auth.Password, "grafanapassword")
 	assert.Equal(t, conf.ExternalServices.Grafana.Auth.Token, "grafanatoken")
 	assert.Equal(t, conf.ExternalServices.Prometheus.Auth.Password, "prometheuspassword")
 	assert.Equal(t, conf.ExternalServices.Prometheus.Auth.Token, "prometheustoken")
 	assert.Equal(t, conf.ExternalServices.Tracing.Auth.Password, "tracingpassword")
 	assert.Equal(t, conf.ExternalServices.Tracing.Auth.Token, "tracingtoken")
 	assert.Equal(t, conf.LoginToken.SigningKey, "signingkey")
-
-	t.Setenv(EnvGrafanaPassword, "grafanapasswordENV")
-	t.Setenv(EnvGrafanaToken, "grafanatokenENV")
-	t.Setenv(EnvPrometheusPassword, "prometheuspasswordENV")
-	t.Setenv(EnvPrometheusToken, "prometheustokenENV")
-	t.Setenv(EnvTracingPassword, "tracingpasswordENV")
-	t.Setenv(EnvTracingToken, "tracingtokenENV")
-	t.Setenv(EnvLoginTokenSigningKey, "signingkeyENV")
-
-	conf, _ = Unmarshal(yamlString)
-
-	// env vars are now set- values should be overridden
-	assert.Equal(t, conf.ExternalServices.Grafana.Auth.Password, "grafanapasswordENV")
-	assert.Equal(t, conf.ExternalServices.Grafana.Auth.Token, "grafanatokenENV")
-	assert.Equal(t, conf.ExternalServices.Prometheus.Auth.Password, "prometheuspasswordENV")
-	assert.Equal(t, conf.ExternalServices.Prometheus.Auth.Token, "prometheustokenENV")
-	assert.Equal(t, conf.ExternalServices.Tracing.Auth.Password, "tracingpasswordENV")
-	assert.Equal(t, conf.ExternalServices.Tracing.Auth.Token, "tracingtokenENV")
-	assert.Equal(t, conf.LoginToken.SigningKey, "signingkeyENV")
```

Comment:
This test is not testing anything anymore. If you want to keep coverage, you can make the base file path internally configurable https://github.com/kiali/kiali/pull/5592/files#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R914 then create a temp testing folder/files with some dummy data in it.

Reply:
I thought about that - but thought it would be adding complexity for the sake of unit testing. Do we do this kind of "internal configuration of file paths for unit tests" anywhere else? I tried to see if there is some Go magic to mock out a file system for tests, but didn't find anything easily consumable.

Reply:
> I thought about that - but thought it would be adding complexity for the sake of unit testing.

This is true but IMO it's worth it. The complexity is small. You'd basically just be putting `/kiali-override-secrets` into a var like `defaultSecretsBasePath` or something and then setting that in the unit test. But another option is to just remove the test which is better than having a test that no longer tests what it says it does.

> I tried to see if there is some Go magic to mock out a file system for tests, but didn't find anything easily consumable.

[t.TempDir()](https://pkg.go.dev/testing#T.TempDir) is what you want. I know we do this elsewhere in the tests but I'd need to track it down.

Reply:
I'll work on this now.

Reply:
@nrfox done. unit tests updated and passing.